### PR TITLE
Fixed container creation

### PR DESCRIPTION
**Pull request by hkaiser**
This patch corrects the containers (`hpx::vector` and `hpx::unordered_map`) to use the new distribution policies to allocate their partitions.

### CODE REVIEW

File: hpx/runtime/components/new.hpp [hunk 1]
```diff
@@ -246,8 +246,24 @@ namespace hpx { namespace components
...
+                    bulk_locality_result;
+
+                hpx::future<std::vector<bulk_locality_result> > f =
+                    policy.template bulk_create<Component>(count,
+                        std::forward<Ts>(vs)...);
+
+                return f.then(launch::sync,
+                    [](hpx::future<std::vector<bulk_locality_result> > && f)
+                    {
+                        std::vector<hpx::id_type> result;
```
Comment (sithhell) [inline hpx/runtime/components/new.hpp]: wouldn't it make sense to call reserve here?

File: hpx/runtime/components/binpacking_distribution_policy.hpp [hunk 2]
```diff
@@ -237,7 +220,29 @@ namespace hpx { namespace components
...
 
-                return detail::consolidate_result(std::move(objs));
+                // consolidate all results
+                return hpx::lcos::local::dataflow(hpx::launch::sync,
+                    [=](std::vector<hpx::future<std::vector<hpx::id_type> > > && v)
+                        mutable -> std::vector<bulk_locality_result>
+                    {
+                        HPX_ASSERT(localities_.size() == v.size());
+
+                        std::vector<bulk_locality_result> result;
```
Comment (sithhell) [inline hpx/runtime/components/binpacking_distribution_policy.hpp]: wouldn't it make sense to call reserve here?

File: hpx/runtime/components/default_distribution_policy.hpp [hunk 3]
```diff
@@ -116,47 +121,67 @@ namespace hpx { namespace components
...
-                objs.push_back(stub_base<Component>::bulk_create_async(
-                    loc, get_num_items(count, loc), vs...));
+                // consolidate all results
+                return hpx::lcos::local::dataflow(hpx::launch::sync,
+                    [=](std::vector<hpx::future<std::vector<hpx::id_type> > > && v)
+                        mutable -> std::vector<bulk_locality_result>
+                    {
+                        HPX_ASSERT(localities_.size() == v.size());
+
+                        std::vector<bulk_locality_result> result;
```
Comment (sithhell) [inline hpx/runtime/components/default_distribution_policy.hpp]: Another candidate for reserving the memory a priori

**Comment by hkaiser** [comment]
This also fixes #1309: hpx::vector partitions are not easily extendable by applications

**Comment by hkaiser** [comment]
@heller: I addressed your comments above

**Pull request merged**